File: army/Unit/Berserk.cpp
#include "Berserk.h"

Berserk::Berserk(std::string nickName, int damage, int hitPoints, std::string title) : Unit(nickName,damage, hitPoints, title) {}

Berserk::~Berserk() {}

void Berserk::addHitPoints(int hp) {
    this->state->addHitPoints(0);
}

void Berserk::takeMagDamage(int dmg) {
    this->state->takeDamage(0);
}

File: army/JavaArmy/src/com/maksklaban/armymax/units/Berserk.java
package com.maksklaban.armymax.units;

import com.maksklaban.armymax.exceptions.*;

public class Berserk extends Unit {
    public Berserk(String nickName, int damage, int hitPoints, String title) throws CloneNotSupportedException {
        super(nickName, damage, hitPoints, title);
    }

    public Berserk(String nickName) throws CloneNotSupportedException {
        this(nickName, 25, 210, "Berserk");
    }

    public void takeMagDamage(int dmg) throws ArmyException {
        this.takeDamage(0);
    }

    public void addHitPoints(int hp) throws ArmyException {
        this.state.addHitPoints(0);
    }
}
